openedx/features/enterprise_support/enrollments/utils.py

```diff
@@ -38,32 +45,72 @@ def lms_enroll_user_in_course(
     Arguments:
      - username (str): User name
      - course_id (obj) : Course key obtained using CourseKey.from_string(course_id_input)
-     - mode (CourseMode): course mode
-     - enterprise_uuid (str): id to identify the enterprise to enroll under
-     - is_active (bool): Optional. A Boolean value that indicates whether the
+     - desired_mode (CourseMode): desired course mode
+     - is_active (bool): A Boolean value that indicates whether the
         enrollment is to be set to inactive (if False). Usually we want a True if enrolling anew.
+     - enterprise_uuid (str): Optional. id to identify the enterprise to enroll under
 
     Returns: A serializable dictionary of the new course enrollment. If it hits
      `CourseEnrollmentExistsError` then it logs the error and returns None.
     """
     user = _validate_enrollment_inputs(username, course_id)
+    current_enrollment = enrollment_api.get_enrollment(username, str(course_id))
 
     with transaction.atomic():
         try:
-            response = enrollment_api.add_enrollment(
+            if current_enrollment and (
```

Comment:
If there _is_ a current enrollment, but the mode and `is_active` fields match the provided arguments, we'll still fall into the `else` block below that adds an enrollment, but I don't think we want to do that.  I think you need one level of `if current_enrollment`, and then a nested branch within there that either updates the existing enrollment, or just returns because everything matches.